src/main/java/duke/Patchi.java

```diff
@@ -0,0 +1,308 @@
+            } else if (input.startsWith("event")) {
+                addEvent(tasks, input);
+            } else {
+                throw new InvalidCommandException();
+            }
+        } catch (MissingDescriptionException e) {
+            println("Patchi: You need to add a task description... Œ(ˊnˋ)B");
+            throw new InvalidCommandException();
+        } catch (MissingTimingException e) {
+            println("Patchi: You need to add a timing for this kind of task! Œ(ˊnˋ)B");
+            throw new InvalidCommandException();
+        } catch (InvalidCommandException e) {
+            println("Patchi: I'm sorry, I don't understand what that means... Œ(ˊnˋ)B");
+            throw new InvalidCommandException();
+        }
+    }
+
+    private static void deleteTask(ArrayList<Task> tasks, String input) throws NoTaskSelectedException {
+        int taskIndex;
+        try {
+            taskIndex = Integer.parseInt(input.substring(7)) - 1;
+        } catch (StringIndexOutOfBoundsException e) {
+            throw new NoTaskSelectedException();
+        }
+
+        if (tasks.size() > taskIndex) {
+            tasks.remove(taskIndex);
+            deleteLine(taskIndex);
+            println("Patchi: Whoosh! I've magicked that task away! Œ(ˊwˋ)B");
+            println("Patchi: You now have " + tasks.size() + " tasks! Œ(ˊwˋ)B");
+        } else {
+            println("Patchi: That task doesn't seem to exist...... Œ(ˊnˋ)B");
+        }
+    }
+
+    private static void addTodo(ArrayList<Task> tasks, String input)
+            throws MissingDescriptionException {
+        String description;
+
+        try {
+            description = input.substring(5);
+        } catch (StringIndexOutOfBoundsException e) {
+            throw new MissingDescriptionException();
+        }
+
+        tasks.add(new Todo(description));
+    }
+
+    private static void addDeadline(ArrayList<Task> tasks, String input)
+            throws InvalidCommandException, MissingTimingException, MissingDescriptionException {
+        int indexOfBy = input.indexOf("/by");
+        String by;
+        String description;
+
+        if (indexOfBy == -1) {
+            throw new InvalidCommandException();
+        }
+
+        try {
+            description = input.substring(9, indexOfBy - 1);
+        } catch (StringIndexOutOfBoundsException e) {
+            throw new MissingDescriptionException();
+        }
+
+        try {
+            by = input.substring(indexOfBy + 4);
+        } catch (StringIndexOutOfBoundsException e) {
+            throw new MissingTimingException();
+        }
+
+        tasks.add(new Deadline(description, by));
+    }
+
+    private static void addEvent(ArrayList<Task> tasks, String input)
+            throws InvalidCommandException, MissingDescriptionException, MissingTimingException {
+        int indexOfAt = input.indexOf("/at");
+        String at;
+        String description;
+
+        if (indexOfAt == -1) {
```

Comment:
Perhaps it would be better to use constants to avoid the use of magic numbers to improve readability